### PR TITLE
chore(Responsive list): migrate to typescript

### PR DESCRIPTION
**Pull request by rivka-ungar**
https://monday.monday.com/boards/3532714909/views/113186840/pulses/4247428543

### CODE REVIEW

File: packages/core/src/components/ResponsiveList/ResponsiveList.tsx [hunk 1]
```diff
@@ -1,37 +1,35 @@
...
-const ResponsiveList = forwardRef(
+const ResponsiveList: VibeComponent<ResponsiveListProps> = forwardRef<HTMLDivElement, ResponsiveListProps>(
   (
     {
-      id,
-      className,
-      rootClassName,
+      id = "",
+      className = "",
+      rootClassName = "",
```
Comment (talkor) [inline packages/core/src/components/ResponsiveList/ResponsiveList.tsx:16]: These and the rest empty strings can just be undefined

File: packages/core/src/components/ResponsiveList/ResponsiveList.tsx [hunk 2]
```diff
@@ -103,44 +101,4 @@ const ResponsiveList = forwardRef(
...
-
-ResponsiveList.menuButtonSizes = MenuButton.sizes;
-ResponsiveList.propTypes = {
-  id: PropTypes.string,
-  className: PropTypes.string,
-  menuButtonClassName: PropTypes.string,
-  menuWrapperClassName: PropTypes.string,
-  /**
-   These attributes will be passed to the MenuButton
-   */
```
Comment (talkor) [inline packages/core/src/components/ResponsiveList/ResponsiveList.tsx:115]: Let's keep those JSDoc comments as they help documenting the props

File: packages/core/src/components/ResponsiveList/ResponsiveList.tsx [hunk 3]
```diff
@@ -74,7 +72,7 @@ const ResponsiveList = forwardRef(
             {directChildren}
             {!!menuChildren.length && (
               <MenuButton
-                componentClassName={cx(styles.listMenuButton, menuButtonClassName)}
+                className={cx(styles.listMenuButton, menuButtonClassName)}
```
Comment (talkor) [inline packages/core/src/components/ResponsiveList/ResponsiveList.tsx:75]: componentClassName was deprecated?

File: packages/core/src/components/ResponsiveList/ResponsiveList.tsx [hunk 4]
```diff
@@ -58,7 +56,7 @@ const ResponsiveList = forwardRef(
     }, [children, index]);
 
     const hiddenChildren = useMemo(() => {
-      const childrenArray = React.Children.toArray(children);
+      const childrenArray = React.Children.toArray(children) as React.ReactElement[];
```
Comment (talkor) [inline packages/core/src/components/ResponsiveList/ResponsiveList.tsx:59]: Can it be a single element?

**Review by talkor — APPROVED** [review]
Excellent 💪🏼 See some minor comments

File: packages/core/src/components/ResponsiveList/ResponsiveList.tsx (hunk 3 above)
Comment (rivka-ungar) [reply packages/core/src/components/ResponsiveList/ResponsiveList.tsx:75]: Seems so
https://github.com/mondaycom/vibe/blob/2ca65624abb07c88fde5e758a47c7e6c4e62b60f/packages/core/src/components/MenuButton/MenuButton.tsx#L30

File: packages/core/src/components/ResponsiveList/ResponsiveList.tsx (hunk 4 above)
Comment (rivka-ungar) [reply packages/core/src/components/ResponsiveList/ResponsiveList.tsx:59]: Even if it's a single element it is converted into an array of 1 item

**Pull request merged**